lib/kino/tree.ex

```diff
@@ -0,0 +1,139 @@
+defmodule Kino.Tree do
+  @moduledoc """
+  A kino for interactively viewing nested data as a tree view.
+
+  The data can be any term.
+
+  ## Examples
+
+      data = %{
+        id: 1,
+        email: "user@example.com",
+        inserted_at: ~U[2022-01-01T10:00:00Z],
```

Comment:
We should specialcase structs like dates and use their inspect, rather than showing a map.

Another option would be to use inspect on structs that implement it (rather than using the default inspect implementation), we would need to make sure multiline items are rendered fine. @josevalim any preference?

Reply:
I think using inspect for structs that implement it is the safest/best choice. Otherwise we can leak a lot of internals, such as in Nx.Tensor. In other words, check if the struct `Inspect.impl_for(data)` returns `Inspect.Any` or anything else.

Reply:
I think grey is fine in this case. We just need to fix the spacing for multiline inspect:

```elixir
%{
  a: "first",
  tensor: Nx.iota({100, 100}),
  z: "last"
}
|> Kino.Tree.new()
```

After that we are ready to go!

Reply:
I think I got it, will push in a moment.

Reply:
<img width="907" alt="image" src="https://user-images.githubusercontent.com/17034772/202854027-a13746b2-a783-4de7-b5d9-8b9e4b3d8e45.png">